### PR TITLE
Upgrade Yarn to 3.6.4 to fix issue with Node 20.x (part 1)

### PR DESCRIPTION
**Pull request by clechasseur**
The TypeScript track currently uses Yarn 3.6.0. However, there is an issue with this version of Yarn when used on recent versions of Node 20.x, as [reported here](https://forum.exercism.org/t/hello-world-did-not-work-on-typescript/8026).

I believe it could be a manifestation of the bug fixed by [this commit](https://github.com/yarnpkg/berry/commit/70cbec62369331c6f75164d66f0f4ca53e5389c2) in Yarn 3.6.3.

This PR updates Yarn to 3.6.4 for all exercises. It also adds a CI step to run a test on the `hello-world` exercise as the student would see it, so that this issue can be detected in the future.

**Note**: the full diff for the changes is actually too big for GitHub, unfortunately, so I cannot create the PR with all changes. The second PR is here: #1407 
The steps to merge all this would be:
- Review and merge this PR; half of the exercises would be updated, as well as all common files; however, pre-checks will fail because the exercises are not all in sync.
- Rebase #1407 to the tip of `main`
- Review and merge #1407 ; all exercises would then be updated and pre-checks will be green.

(I am submitting this following @iHiD 's request on the Discord server. I will post on the forum when both PRs are ready.)

### CODE REVIEW

File: .github/workflows/ci.js.yml [hunk 1]
```diff
@@ -43,3 +43,34 @@ jobs:
...
+          rm -f hello-world.ts
+          cp ./.meta/proof.ci.ts hello-world.ts
+
+      - name: Install project dependencies
+        working-directory: exercises/practice/hello-world
+        run: yarn install --no-immutable
+
+      - name: Run tests
+        working-directory: exercises/practice/hello-world
+        run: yarn test
```
Comment (ErikSchierboom) [inline .github/workflows/ci.js.yml:76]: Could you extract this to a separate PR? It is related to this PR, but it is something else IMHO.

File: .github/workflows/pr.ci.js.yml [hunk 2]
```diff
@@ -53,3 +53,34 @@ jobs:
...
+          rm -f hello-world.ts
+          cp ./.meta/proof.ci.ts hello-world.ts
+
+      - name: Install project dependencies
+        working-directory: exercises/practice/hello-world
+        run: yarn install --no-immutable
+
+      - name: Run tests
+        working-directory: exercises/practice/hello-world
+        run: yarn test
```
Comment (ErikSchierboom) [inline .github/workflows/pr.ci.js.yml:86]: Could you extract this to a separate PR? It is related to this PR, but it is something else IMHO.

**Review by ErikSchierboom — COMMENTED** [review]
I've left some comments. Before we can merge this, I'd like to verify that this works with the test runner (do we need to upgrade there). Let me know if you need help with that

**Comment by clechasseur** [comment]
I've moved the new CI step to another PR: #1409 

> I've left some comments. Before we can merge this, I'd like to verify that this works with the test runner (do we need to upgrade there). Let me know if you need help with that

As I mentioned on the forum, I tried to make it work earlier this week but got a weird script error - but I admit I didn't have time to investigate. To save you time, let me take another stab at it first.

Would we want to update the test runner to Node 20 if we also update Yarn there?

**Comment by clechasseur** [comment]
@ErikSchierboom Ok, I've finally been able to sort my script issue.

When I run the `typescript-test-runner` locally, it works fine:

Output: [bin_run_sh_output.log](https://github.com/exercism/typescript/files/13331005/bin_run_sh_output.log)
Result JSON: [results.json](https://github.com/exercism/typescript/files/13331007/results.json)

I've tried this both with Node 18 and Node 20 and I get the same result for both (a success).

I think the test runner works fine because it's not using PnP like is used when a student downloads an exercise locally.

Would you like me to try to update Yarn to 3.6.4 in the test runner (and perhaps Node to 20.x)? In any case, the test runner is working correctly as-is so it could be updated later if preferred.

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
> Would you like me to try to update Yarn to 3.6.4 in the test runner (and perhaps Node to 20.x)? In any case, the test runner is working correctly as-is so it could be updated later if preferred.

Yeah that would be nice!

**Comment by clechasseur** [comment]
> > Would you like me to try to update Yarn to 3.6.4 in the test runner (and perhaps Node to 20.x)? In any case, the test runner is working correctly as-is so it could be updated later if preferred.
> 
> Yeah that would be nice!

@ErikSchierboom PR for the test runner update: exercism/typescript-test-runner#65
